sklearn_numba_dpex/common/tests/test_kernels.py

```diff
@@ -0,0 +1,92 @@
+import pytest
+
+import dpctl
+import dpctl.tensor as dpt
+
+import numpy as np
+
+from sklearn_numba_dpex.common.kernels import (
+    make_sum_reduction_2d_axis1_kernel,
+    make_argmin_reduction_1d_kernel,
+)
+from sklearn_numba_dpex.testing.config import float_dtype_params
+
+
+@pytest.mark.parametrize("dtype", float_dtype_params)
+def test_sum_reduction_2d(dtype):
+    n_reductions = 3
+    n_items = 4
+    device = dpctl.SyclDevice()
+    work_group_size = device.max_work_group_size
```

Comment:
You can use random data values from numpy that you move to the sycl device via `dpt` instead of using `dpt.reshape(dpt.arange(12, dtype=dtype), (3, 4))` and then compare to the reduction results computed by numpy.

Reply:
Why use random data values for a sum/argmin ? I can't think of specific cases that we could miss with hard-coded trivial data, and it's not readable. Plus those tests can be a little expensive because there's a jit for each (n_reduction, n_items) combination.

OK for tests that highlight what happens with invalid combinations but I don't think it's very important either since those functions are not intended to be public but only have use in top level functions that are also tested.

Reply:
Also when work_group_size is not a power of 2 the behavior should be undefined. Could work, could fail. could segfault. Do you want that the instantiation function perform some input validation and raise error if `work_group_size` is not a power of 2 ?

Reply:
I think it's should be quite cheap to check yes + it's protected by the lru cache.

Reply:
> OK for tests that highlight what happens with invalid combinations but I don't think it's very important either since those functions are not intended to be public but only have use in top level functions that are also tested.

Regular data or deterministically RNG'ed data as you prefer. My point it to do some checks with larger dimensions than `workgroup_size` and compare to the results of the same reduction computed by numpy.

Reply:
OK for validating `work_group_size`, it's actually very cheap.

I'll parametrize work groups and and a use case with shape of axis 1 set to length 5 to test cases where work group size and axis sizes do not divide each other